pragma/tests/fetchers_test.py

```diff
@@ -263,12 +279,17 @@ async def test_async_index_fetcher(fetcher_config, mock_data, forked_client):
 async def test_async_index_fetcher_404(fetcher_config, mock_data, forked_client):
     # we only want to mock the external fetcher APIs and not the RPC
     with aioresponses(passthrough=[forked_client.client.url]) as mock:
-        fetcher = fetcher_config["fetcher_class"](SAMPLE_ASSETS, PUBLISHER_NAME)
+        sample_assets = [
+            SAMPLE_ASSETS[i]
+            for i in range(len(SAMPLE_ASSETS))
+            if SAMPLE_ASSETS[i]["type"] == "SPOT"
+        ]
```

Comment:
Explain in the comment why this change and create a function filter_assets for example